### ISSUE TITLE
Pass backend_config to the init command on apply job

### ISSUE THREAD

**Issue by toddpla** [CircleCI-Public/terraform-orb#8]
**Orb Version**
1.0.

**Describe the bug**
Trying to use the `terraform/apply` with a custom backend_config like so:
```yml
workflows:
  branch_push:
    jobs:
      - terraform/apply:
          checkout: true
          path: ./terraform/envs/generic
          backend_config: 'bucket=some-bucket,key=development/some-appi/terraform.tfstate'
          context: API_BUILD
```
But this throws:
<img width="712" alt="Screenshot 2020-10-14 at 23 42 03" src="https://user-images.githubusercontent.com/31185699/96053190-f9836600-0e76-11eb-90c3-82b6d6f4b458.png">

**To Reproduce**

1. Add a `terraform/apply` job to a workflow with backend_config
2. Start workflow
3. Error throws

**Expected behavior**

I expect the terraform to apply presuming that the backend_config would be passed to the init command within the apply job.

**Additional context**

Would it work to pass the backend_config parameter through [here](https://github.com/CircleCI-Public/terraform-orb/blob/master/src/jobs/apply.yml#L63).

**Comment by aedifex** [comment]
Thanks for bringing this up @toddpla ! Your thoughts on expected behavior are spot on. I believe we're going to include this fix in a pretty significant refactor in the coming weeks. We should be pushing an updated Orb version in the next day or so which will include a "workaround", that will hopefully address the issue in the interim. A significant refactor is planed in the coming weeks. As always, we're actively soliciting feedback and thoughts from the community so if anything else comes up or comes to mind, please let us know!

**Comment by KyleTryon** [comment]
This has since been resolved in #10. 3.0.0 of the orb will be being released soon in #46
